[Sig Events] Add architecture and operator docs (#7497)
elastic/docs-content: solutions/observability/streams/significant-events/operator-guide.md

---
navigation_title: Operator guide
description: System impact, cost drivers, and operational procedures for cluster operators running Significant Events.
applies_to:
  serverless: beta
  stack: beta 9.5+
products:
  - id: observability
  - id: elasticsearch
  - id: kibana
  - id: cloud-serverless
  - id: cloud-hosted
  - id: cloud-enterprise
  - id: cloud-kubernetes
  - id: elastic-stack
---

# Operator guide [sig-events-operator]

Use this guide to understand how Significant Events affects your cluster and how to manage it.

> - [What runs where](#sig-events-op-components): which components run on {{kib}}, {{es}}, and Workflows

Reviewer: ```suggestion
- [What runs where](#sig-events-op-components): Which components run on {{kib}}, {{es}}, and Workflows
```

> - [System impact](#sig-events-op-impact): query load, pipeline lag, memory, and storage growth

Reviewer: ```suggestion
- [System impact](#sig-events-op-impact): Query load, pipeline lag, memory, and storage growth
```

- [Cost drivers](#sig-events-op-costs): LLM call sites and token usage by phase

> - [Disable and re-enable](#sig-events-op-disable): how to stop the pipeline or pause KI refresh

Reviewer: ```suggestion
- [Disable and re-enable](#sig-events-op-disable): How to stop the pipeline or pause Knowledge indicator (KI) refresh
```

> - [Recovery procedures](#sig-events-op-recovery): symptoms and actions for common degraded states

Reviewer: ```suggestion
- [Recovery procedures](#sig-events-op-recovery): Symptoms and actions for common degraded states
```

## What runs where [sig-events-op-components]

The following table shows each pipeline component, where it runs, what triggers it, and what it reads and writes:

> | Component | Runs on | Trigger | Reads | Writes |

Reviewer: Since the second column contains a mix of Solutions and features, a more general term like "Uses" might be better than "Runs on". I'd also recommend sticking with either solutions or features, not both. The following suggestions use features.   

```suggestion
| Component | Uses | Trigger | Reads | Writes |
```

|---|---|---|---|---|

> | Knowledge indicator (KI) feature identification | {{kib}} (Task Manager + workflow) | On-demand / continuous extraction | Stream logs | `.significant_events-knowledge_indicators` |

Reviewer: I'm not sure how well "On-demand / continuous extraction" fits in the Trigger column, which describes what triggers the pipeline component.

```suggestion
| KI feature identification  | Task Manager + Workflow | On-demand / continuous extraction | Stream logs | `.significant_events-knowledge_indicators` |
```

> | KI query generation | {{kib}} (workflow) | On-demand | Features + existing queries | Query KI assets |

Reviewer: Same comment as above - I'm not sure "On-demand" works in the Trigger column.

```suggestion
| KI query generation | Workflow | On-demand | Features + existing queries | Query KI assets |
```

> | Alerting rule execution | {{kib}} alerting → {{es}} | Per-rule schedule | Stream data via {{esql}} | `.rule-events` |

Reviewer: ```suggestion
| Alerting rule execution | {{kib}} alerting → {{es}} | Per-rule schedule | Stream data using {{esql}} | `.rule-events` |
```

> | Detection workflow | {{kib}} Workflows | Cron 10m | `.rule-events` | `.significant_events-detections` |

Reviewer: ```suggestion
| Detection Workflow | Workflow | Cron 10m | `.rule-events` | `.significant_events-detections` |
```

> | Discovery workflow | {{kib}} Workflows + Agent Builder | Cron 10m | `.significant_events-detections` + KIs | `.significant_events-discoveries` |

Reviewer: ```suggestion
| Discovery Workflow | Workflows + Agent Builder | Cron 10m | `.significant_events-detections` + KIs | `.significant_events-discoveries` |
```

> | Discovery workflow | {{kib}} Workflows + Agent Builder | Cron 10m | `.significant_events-detections` + KIs | `.significant_events-discoveries` |

Reviewer 2: suggestion: the table lists Detection and Discovery workflows but not Triage.

The triage/judge workflow (`system-significant-events-triage`, agent `significant_events.discovery-judge`) is a real third workflow and it's the thing that actually writes Significant Events. Add a row:

```
| Triage workflow | {{kib}} Workflows + Agent Builder | Cron 10m | `.significant_events-discoveries` | `.significant_events-events` |
```

## System impact [sig-events-op-impact]

The following sections describe the query load, pipeline lag, memory, and storage growth you can expect when running Significant Events. These are observable signals, not hard guarantees.

**Alerting rule query load**

> One {{esql}} alerting rule runs per promoted query KI. Each rule fires on its own schedule. A `change_point` aggregation runs across all active rules' alert results on each detection cycle. Alert volume in `.rule-events` grows proportionally to the number of promoted rules and the rate at which they fire.

Reviewer: ```suggestion
One {{esql}} alerting rule runs for each promoted query KI. Each rule executes on its own schedule, creating an alert when its query matches. On each detection cycle, change point aggregation runs separately for each active rule, against that rule's per-time-bucket alert counts. Document volume in `.rule-events` grows with the number of promoted rules and how often they execute.
```

**Pipeline lag**

The following table provides illustrative estimates of how long each phase takes from trigger to output, based on workflow schedules:

| Phase | Typical lag |
|---|---|
| First detection | ~10 min |
| Discovery | ~10 min after detection |
| New Significant Event | Sync with discovery |
| Stale re-review | Up to ~10 min |

**{{kib}} memory**

> Agent runs add memory pressure. Observe {{kib}} memory under realistic load. Exact memory requirements depend on your workload. Measure under realistic load before committing to a configuration.

Reviewer: Flagging this because I'm not sure if "Agent" refers to the Judge and Discovery agents, AI agents, or LLMs.

**Ingest assumptions**

Streams must have recent log data for KI extraction to produce useful results. Empty streams or streams with very sparse data produce weak KIs, which produce weak query KIs, which produce fewer promoted rules.

**Storage growth**

Significant Events writes to the following data streams:

| Data stream | Written by | Growth driver |
|---|---|---|

> | `.significant_events-detections` | Detection workflow | Append-only; one document per observed state transition per rule |

Reviewer: ```suggestion
| `.significant_events-detections` | Detection Workflow | Append-only; one document per observed state transition per rule |
```

> | `.significant_events-discoveries` | Discovery agent + Judge agent | Append-only; one document per discovery state change |

Reviewer 2: nitpick (non-blocking): the judge doesn't write to the discoveries stream.

`.significant_events-discoveries` is written by the discovery agent (`discovery_write`) and by the triage workflow's deterministic `handled` markers — but the judge agent itself only writes to `.significant_events-events` (`events_write`). "Discovery agent + Judge agent" for the discoveries row is slightly off; the judge belongs on the events row (which you already have right).

```suggestion
| `.significant_events-discoveries` | Discovery agent | Append-only; one document per discovery state change |
```

| `.significant_events-events` | Judge | Append-only; one document per Significant Event state change |

> The `.significant_events-*` data streams use Data Stream Lifecycle (DSL) with a default 90-day retention. You can override retention per stream using the DSL API. See [{{esql}} traceability](./how-it-works.md#sig-events-hiw-traceability) for the full index layout and traceability guidance.

Reviewer 2: suggestion: "the `.significant_events-*` data streams use DSL with a default 90-day retention" isn't uniform.

Detections and discoveries do set `lifecycle: { data_retention: '90d' }`, but the events data stream template has *no* lifecycle block at all (`events/data_stream.ts`), so it doesn't get the 90-day default. Given events are the highest-value artifact, this is a real gap worth being precise about:

```suggestion
The `.significant_events-detections` and `.significant_events-discoveries` data streams use DSL with a default 90-day retention. (The events data stream currently has no default retention configured.) You can override retention per stream using the DSL API. See [{{esql}} traceability](./how-it-works.md#sig-events-hiw-traceability) for the full index layout and traceability guidance.
```

| Query generation | Per stream after features exist | Medium; requires reasoning and ES\|QL validation |
| Discovery agent | ~10 min cycles when unprocessed detections exist | Bursty; scales with the number of active alerting rules firing |
| Judge agent | Sync on new discoveries; stale re-review ~10 min | Lower frequency; scales with the number of open discoveries |

Continuous extraction is the largest cost multiplier. When enabled, feature identification runs on a recurring schedule across all eligible streams. Enabling it on a large number of streams significantly increases token consumption.

<!-- Billing policy still needs to be defined/documented -->

## Disable and re-enable [sig-events-op-disable]

### Disable Significant Events entirely

Set `observability:streamsSigEventsScheduledDiscoveryEnabled` to `false` in {{kib}} settings. This stops the detection, discovery, and triage workflows.

### Stop background KI refresh only

To stop continuous extraction without disabling Significant Events:

1. Select **Significant Events** → **Settings**.
2. Under **Continuous KI extraction**, turn off **Enable continuous KI extraction**.

> Disabling continuous extraction cancels all in-flight feature identification tasks and force-deletes the continuous extraction workflow. Already-extracted KIs are not deleted. Manually-triggered extractions continue to work.

Reviewer 2: todo: same force-delete wording as `knowledge-indicators.md`. The managed workflow is disabled, not force-deleted.

```suggestion
Disabling continuous extraction cancels all in-flight feature identification tasks and disables the continuous extraction workflow. Already-extracted KIs are not deleted. Manually-triggered extractions continue to work.
```
